bigtable/client/rpc_backoff_policy.cc

```diff
@@ -55,5 +55,28 @@ std::chrono::milliseconds ExponentialBackoffPolicy::on_completion(
   return delay;
 }
 
+std::unique_ptr<RPCBackoffPolicy> RandomizedBackoffPolicy::clone() const {
+  return std::unique_ptr<RPCBackoffPolicy>(new RandomizedBackoffPolicy(*this));
+}
+
+void RandomizedBackoffPolicy::setup(grpc::ClientContext& /*unused*/) const {}
+
+std::chrono::milliseconds RandomizedBackoffPolicy::on_completion(
+    grpc::Status const& status) {
+  using namespace std::chrono;
+  auto delay = duration_cast<milliseconds>(current_delay_);
+  std::random_device rd;
+  std::mt19937 rng_generator(rd());
+  auto min_value = duration_cast<microseconds>(current_delay_).count();
+  auto max_value = duration_cast<microseconds>(maximum_delay).count();
+  std::uniform_int_distribution<int> rng_distribution(min_value, max_value);
```

Comment:
This is an interesting approach, but not one I was familiar with.  Typically one combines the exponential backoff **and** randomization, so the code becomes:

```C++
std::uniform_int_distribution<int> distribution(0, duration_cast<microseconds>(current_delay_).count());
auto delay = microseconds(distribution(generator));
current_delay_ *= 2;
if (current_delay_ >= maximum_delay_) {
  current_delay_ = maximum_delay_;
}
return delay;
```

Reply:
Done.